presto-rcfile/src/main/java/io/prestosql/rcfile/text/TimestampEncoding.java

```diff
@@ -113,11 +127,16 @@ public Block decodeColumn(ColumnData columnData)
     @Override
     public void decodeValueInto(int depth, BlockBuilder builder, Slice slice, int offset, int length)
     {
-        type.writeLong(builder, parseTimestamp(slice, offset, length));
+        DecodedTimestamp decodedTimestamp = parseTimestamp(slice, offset, length);
+        prestoTimestampEncoder.write(decodedTimestamp, builder);
     }
 
-    private static long parseTimestamp(Slice slice, int offset, int length)
+    private static DecodedTimestamp parseTimestamp(Slice slice, int offset, int length)
     {
-        return HIVE_TIMESTAMP_PARSER.parseMillis(new String(slice.getBytes(offset, length), US_ASCII)) * MICROSECONDS_PER_MILLISECOND;
+        long millis = HIVE_TIMESTAMP_PARSER.parseMillis(new String(slice.getBytes(offset, length), US_ASCII));
```

Comment:
I guess this should parse with higher precision
(unless Hive is not able to write with precvision higher than millis, then a comment would be needed)

Are we missing product tests covering all timestamp precisions x file formats?

Reply:
Ugh, I was under the impression that `RCTEXT` only supports millisecond precision.  I'll fix.

Reply:
don't take my word for this.